subprojects/core/src/main/java/org/gradle/internal/classpath/InstrumentingTransformer.java

```diff
@@ -272,6 +310,15 @@ public void visitInvokeDynamicInsn(String name, String descriptor, Handle bootst
                 super.visitInvokeDynamicInsn(name, descriptor, bootstrapMethodHandle, bootstrapMethodArguments);
             }
         }
+
+        private boolean isGradleLambdaDescriptor(String descriptor) {
+            return isLambdaDescriptorOf(Action.class, descriptor)
+                || isLambdaDescriptorOf(Spec.class, descriptor);
+        }
+
+        private boolean isLambdaDescriptorOf(Class<?> clazz, String descriptor) {
+            return descriptor.endsWith(")" + getType(clazz).getDescriptor());
```

Comment:
maybe turn this into some constants, as this is called many times.

Reply:
Done.